session/types.proto

```diff
@@ -232,3 +278,114 @@ message ResponseVerificationHeader {
   // Chain of previous hops signatures
   ResponseVerificationHeader origin = 4 [json_name = "origin"];
 }
+
+// Session Token v2
+
+// Target account for SessionTokenV2.
+// It can be either direct (OwnerID) or indirect (NNS domain).
+message Target {
+  // Account identifier.
+  oneof identifier {
+    // Direct account reference via OwnerID (hash of verification script).
+    neo.fs.v2.refs.OwnerID owner_id = 1 [json_name = "ownerID"];
+
+    // Indirect account reference via NeoFS Name Service.
+    // NNS name is a domain name that resolves to an OwnerID through the
+    // NeoFS Name Service. The name must be a valid DNS-like domain name
+    // (e.g., "example.neofs") that is registered in the NNS contract on
+    // the Neo blockchain. The NNS record should contain a string record with
+    // the corresponding OwnerID value in NEP-18 address format.
+    string nns_name = 2 [json_name = "nnsName"];
+  }
+}
+
+// Lifetime parameters of the token v2. Field names taken from rfc7519.
+message TokenLifetime {
+  // Expiration, the last valid Unix timestamp.
+  uint64 exp = 1 [json_name = "exp"];
+
+  // Not valid before, the first valid Unix timestamp.
+  uint64 nbf = 2 [json_name = "nbf"];
+
+  // Issued at, the Unix timestamp when the token was issued.
+  uint64 iat = 3 [json_name = "iat"];
+}
+
+
+// DelegationInfo represents a single delegation in a chain of trust.
+message DelegationInfo {
```

Comment:
We need some "final" bit as well for the case where token can't be used to build longer chains (A trusts B to do something, but doesn't give B permission to issue tokens for C).